wavplayer.cpp

```diff
@@ -727,18 +755,19 @@ DWORD WINAPI downloadAudioProc(LPVOID lpParam){
             soundTouch, 
             wavSpec.format,
             wavSpec.freq, 
-            audioParams->sampleRate, 
-            !filterApplied);
+            audioParams->sampleRate,
+            audioParams->numChannels,
+            true);
```

Comment:
we always need convert to f32 audio data at this point (and if we're not pitch shifting, the `extractAudioData` function will do so as well), hence the `true` here. 

any subsequent filters will be using f32 data so we know for sure that no more audio data conversion should be needed, hence the `false`'s for the below filters.